### DISCUSSION TITLE
Faster ec.encode with performant temp space

### DISCUSSION THREAD

**Discussion by bradmurray** [seaweedfs/seaweedfs#6230] [Ideas]
While an ec.encode is always going to be limited by the read speed of the drive where the data sits, wouldn't it be much faster if the .ec## files were written in a temporary space or possibly even directly to other volume servers? It seems that the biggest part of a slow encode is that the volume being encoded is also responsible for writing all 14 files which then have to be distributed. I haven't tried it yet, but I assume I would get better encoding by moving the entire volume to an SSD and then doing the encode there and then distributing it to the HDD volumes.

**Comment by chrislusf** [comment]
there is a PR addressing this. https://github.com/seaweedfs/seaweedfs/pull/6111

@sek2022 do you want to continue with this?

**Reply by sek2022** [reply]
> there is a PR addressing this. #6111
> 
> @sek2022 do you want to continue with this?

Yes, I'm testing. I'll commit it after soon.
You can test this branch first, @bradmurray 
https://github.com/sek2022/seaweedfs

**Reply by bradmurray** [reply]
> You can test this branch first, @bradmurray https://github.com/sek2022/seaweedfs

I use largedisk. I also don't have a test env to toy with and I'm already up to v3.80 on my cluster.
